Source/Core/Core/PowerPC/PowerPC.cpp

```diff
@@ -323,6 +323,66 @@ void SingleStep()
   s_cpu_core_base->SingleStep();
 }
 
+bool StepOut(std::optional<std::chrono::seconds> timeout)
+{
+  breakpoints.ClearAllTemporary();
+
+  // Keep stepping until the next return instruction or timeout
+  using clock = std::chrono::steady_clock;
+  clock::time_point clock_end = clock::now();
+  if (timeout)
+    clock_end += timeout.value();
+  CoreMode old_mode = GetMode();
+  SetMode(CoreMode::Interpreter);
+
+  // Loop until either the current instruction is a return instruction with no Link flag
+  // or a breakpoint is detected so it can step at the breakpoint. If the PC is currently
+  // on a breakpoint, skip it.
+  UGeckoInstruction inst = HostRead_Instruction(PC);
+  do
+  {
+    if (WillInstructionReturn(inst))
+    {
+      SingleStep();
+      break;
+    }
+
+    if (inst.LK)
+    {
+      // Step over branches
+      u32 next_pc = PC + 4;
+      do
+      {
+        SingleStep();
+      } while (PC != next_pc && (!timeout || clock::now() < clock_end) &&
+               !breakpoints.IsAddressBreakPoint(PC));
+    }
+    else
+    {
+      SingleStep();
+    }
+
+    inst = HostRead_Instruction(PC);
+  } while ((!timeout || clock::now() < clock_end) && !breakpoints.IsAddressBreakPoint(PC));
+
+  SetMode(old_mode);
+  if (timeout && clock::now() >= clock_end)
```

Comment:
Wouldn't this just be:

```cpp
return !timeout || clock::now() < clock_end;
```

which is used in around two other places in the function, which may or may not make it desirable to wrap it into a lambda like:

```cpp
auto some_descriptive_name = [&]{ return !timeout || clock::now() < clock_end; }
```

and just be called in place of those occurrences?